x-pack/plugins/security_solution/public/management/hooks/use_with_show_responder.tsx

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback } from 'react';
+import type { ConsoleResponseActionCommands } from '../../../common/endpoint/service/response_actions/constants';
+import { isResponseActionSupported } from '../../../common/endpoint/service/response_actions/is_response_action_supported';
+import { HeaderSentinelOneInfo } from '../components/endpoint_responder/components/header_info/sentinel_one/header_sentinel_one_info';
+
+import { useUserPrivileges } from '../../common/components/user_privileges';
+import {
+  ActionLogButton,
+  getEndpointConsoleCommands,
+  OfflineCallout,
+} from '../components/endpoint_responder';
+import { useConsoleManager } from '../components/console';
+import { MissingEncryptionKeyCallout } from '../components/missing_encryption_key_callout';
+import type { SentinelOneAgentInfo } from '../../../common/types';
+import { RESPONDER_PAGE_TITLE } from './translations';
+import { getCommandKey } from '../components/endpoint_response_actions_list/components/hooks';
+
+type ShowResponseActionsConsole = (agentInfo: SentinelOneAgentInfo) => void;
+export const useWithShowResponder = (): ShowResponseActionsConsole => {
+  const consoleManager = useConsoleManager();
+  const endpointPrivileges = useUserPrivileges().endpointPrivileges;
+
+  return useCallback(
+    (agentInfo: SentinelOneAgentInfo) => {
+      // If no authz, just exit and log something to the console
+      if (!endpointPrivileges.canAccessResponseConsole) {
+        window.console.error(
+          new Error(`Access denied to ${agentInfo.agent.type} response actions console`)
+        );
+        return;
+      }
+
+      const endpointAgentId = agentInfo.agent.id;
+      const endpointRunningConsole = consoleManager.getOne(endpointAgentId);
+
+      if (endpointRunningConsole) {
+        endpointRunningConsole.show();
+      } else {
+        consoleManager
+          .register({
+            id: endpointAgentId,
+            meta: {
+              sentinel_one: agentInfo,
+            },
+            consoleProps: {
+              commands: getEndpointConsoleCommands({
+                agentType: agentInfo.agent.type,
+                endpointAgentId,
+                endpointCapabilities: ['isolation'],
+                endpointPrivileges,
+              }).filter((command) =>
```

Comment:
I don't think you want to filter, but rather you want to `.map()` and for those commands that are not supported, you want to change the command definition to:

```typescript
{
  ...command,
  helpHidden: true,
  about: {
    ...command.about,
    isSupported: false,
  }
}
```


You should then be able to enter a command that is not supported (ex. `execute`) and the console should report an error that alludes that the command is "known" but not supported (we may need to change the error message)

Reply:
@paul-tavares Is it something like this that we should see instead of filtered commands?
![Screenshot 2024-01-09 at 9 01 20 PM](https://github.com/elastic/kibana/assets/1849116/0fcdb183-c45a-4fb3-85e4-aac82556c7fc)

Reply:
No, we should not see the commands in that list. There is a separate property in the Command definition that hides it from the help menu.